libs/payments/ui/src/lib/client/components/CouponForm/index.tsx

```diff
@@ -60,6 +60,30 @@ const WithCoupon = ({
   );
 };
 
+const CouponInput = ({
```

Comment:
I've had some chats with @StaberindeZA about breaking out components, and the general result has been keeping things together in one single component/page wherever possible. Can this be merged back into its parent component, or does it need to be split out for form management?

Reply:
I don't think this can be merged back into its parent component if we want to use `pending` in `useFormStatus`.

Per [React documentation for `useFormStatus`](https://react.dev/reference/react-dom/hooks/useFormStatus), it's considered a pitfall - `useFormStatus` will not return status information for a `<form>` rendered in the same component.

Reply:
TIL. I didn't know that about `useFormStatus`. I'm happy to keep as to keep `useFormStatus`.